src/web/mjs/engine/Connector.mjs

```diff
@@ -476,6 +476,28 @@ export default class Connector {
             } );
     }
 
+    fetchGraphQL(request, operationName, query, variables) {
```

Comment:
Since the goal is to customize the request options, the `Object.assign` method could be utilized, which requires some changes to the code.

Reply:
Is this method general purpose to be added to the base class (e.g. will it work for GQL queries without variables => `undefined`)?
Could this directly be used in other websites, e.g. `Luscious`, `MangaHub` or `ReaderFront` without any further adjustments?

If not, consider to move it into the specific class, or made it generic enough to meet the requirements so it could be used in other websites in the future.

Reply:
It's generic, but only for graphQL JSON requests (https://graphql.org/learn/serving-over-http/#post-request). Not sure if all website accept json requests.

I'll move it to the specific class, it's safer

Reply:
Or I can rename it `fetchGraphQLOverJson()`. What do you prefer ?

Reply:
Using the modifications from the last commit, I have tested it on `Luscious`, `MangaHub` and `ReaderFront` and it works on all of them.
I can make another PR if you want for them if you want

Reply:
Sounds good